feat: fcc-todo-app (#51367)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64fb1c4dc0feb219149a7c7d.md

---
id: 64fb1c4dc0feb219149a7c7d
title: Step 46
challengeType: 0
dashedName: step-46
---

# --description--

> If the user attempts to edit a task and decides not to make changes before closing the form, you don't need to show them the modal containing the `Cancel` and `Discard` buttons.

Reviewer: ```suggestion
If the user attempts to edit a task but decides not to make any changes before closing the form, there is no need to display the modal with the `Cancel` and `Discard` buttons.
```

> Inside the `closeTaskFormBtn` event listener, use `const` to create another variable named `formInputValuesUpdated` to check if the user made changes while trying to edit a task. For the value, check whether `titleInput` value **is not equal** to `currentTask.title`, whether `dateInput` value **is not equal** to `currentTask.date`, and whether `descriptionInput` value **is not equal** to `currentTask.description`.

Reviewer: ```suggestion
Inside the `closeTaskFormBtn` event listener, use `const` to create another variable named `formInputValuesUpdated`. Check if the user made changes while trying to edit a task by verifying that the `titleInput` value **is not equal to** `currentTask.title`, the `dateInput` value **is not equal to** `currentTask.date`, and the `descriptionInput` value **is not equal to** `currentTask.description`.
```
